### DISCUSSION TITLE
How do I make field's value autogenerate?

### DISCUSSION THREAD

**Discussion by PrashantDhungana** [filamentphp/filament#6965] [Help] [answered]
I have the following text input. I need to generate an incremental code for a new property record. (E.g: If the previous code was "00000" then the next should be automatically filled up as "00001".) 
The following code works for Create Page but doesn't work for Edit Page.

For the Edit Page too, the field is incremented by 1.
`
TextInput::make('code')
                    ->required()
                    ->afterStateHydrated(function (TextInput $component, $state) {
                        $lastProperty = Property::latest()->first();
                        $lastCode = $lastProperty ? $lastProperty->code : '00000';
                        $component->state($lastCode + 1);
                    })
                    ->unique()
                    ->length(8)
                    ->placeholder('PINXXXXX')
                    ->label('Property Code'),
`

**Comment by pboivin** [comment]
I'm curious, why do you need this on an Edit page? Maybe I'm missing something but it looks like this would always take the latest code and increment by 1, every time someone edits a record?

**Reply by PrashantDhungana** [reply]
Thank you for the reply and also sorry for the confusion. Let me explain you in detail. I have PropertyResource. In the create form, I need a field "code" that should generate a property code (XXXXX). This code shouldn't be random but generated based on the previous property code (incremented by 1). 

The code I provided works for Create form. But on Edit form too, the property code is incremented by 1, which I don't want to. I only want to generate the incremented code on Create page but not on Edit page.

**Reply by pboivin** [reply, marked as answer]
Thanks for the details, it makes sense. I think this should work :
```php
->afterStateHydrated(function ($state, $component) {
    if ($state) {
        return;
    }
    
    // state has no value, initialize...
}),
```

**Reply by PrashantDhungana** [reply]
It worked! Thanks alot. You are awesome
